internal/core/layout.rs

```diff
@@ -401,6 +401,23 @@ mod grid_internal {
     }
 }
 
+static mut GLOBAL_TEST_MODE: bool = false;
```

Comment:
Let's not use `static mut`

```suggestion
static GLOBAL_TEST_MODE: core::sync::atomic::AtomicBool = core::sync::atomic::AtomicBool::new(false);
```

Then you don't need unsafe.

Reply:
Code removed.